### PR TITLE
Asynchronous session handling

### PR DESCRIPTION
**Pull request by sonnyp**
Emit an event on BOSHServer for every new session creation with the session id as arugment and a callback that will accept and process the session creation when it's called.

### CODE REVIEW

File: lib/bosh/http.js [hunk 1]
```diff
@@ -158,6 +160,8 @@ BOSHServer.prototype._createSession = function(req, res, bodyEl) {
     var session = new BOSHServerSession({ req: req, res: res, bodyEl: bodyEl , options: options})
     this.sessions[session.sid] = session
```
Comment (sonnyp) [inline lib/bosh/http.js]: TODO: move this to the callback

**Comment by mllocs** [comment]
description please

**Comment by csaura** [comment]
looks good

**Comment by mllocs** [comment]
:+1:

**Pull request merged**